govcd/vm_dhcp_test.go

```diff
@@ -0,0 +1,228 @@
+	check.Assert(ips, HasLen, 2)
+	check.Assert(ips[0], Equals, "")
+	check.Assert(ips[1], Equals, "")
+
+	if testVerbose {
+		fmt.Println("OK")
+	}
+
+	// err = vm.PowerOnAndForceCustomization()
+	task, err = vapp.PowerOn()
+	check.Assert(err, IsNil)
+	err = task.WaitTaskCompletion()
+	check.Assert(err, IsNil)
+
+	if testVerbose {
+		fmt.Printf("# Get IPs for NICs 0 and 1: ")
+	}
+	// Wait and check DHCP lease acquired
+	ips, hasTimedOut, err = vm.WaitForDhcpIpByNicIndexes([]int{0, 1}, 300, true)
+	check.Assert(err, IsNil)
+	check.Assert(hasTimedOut, Equals, false)
+	check.Assert(ips, HasLen, 2)
+	check.Assert(ips[0], Matches, `^32.32.32.\d{1,3}$`)
+	check.Assert(ips[1], Matches, `^32.32.32.\d{1,3}$`)
+
+	if testVerbose {
+		fmt.Printf("OK:(NICs 0 and 1): %s, %s\n", ips[0], ips[1])
+	}
+
+	// DHCP lease was received so VMs MAC address should have an active lease
+	if testVerbose {
+		fmt.Printf("# Get active lease for NICs with MAC 0: ")
+	}
+	lease, err := edgeGateway.GetNsxvActiveDhcpLeaseByMac(netCfg.NetworkConnection[0].MACAddress)
+	check.Assert(err, IsNil)
+	check.Assert(lease, NotNil)
+	check.Assert(lease.IpAddress, Matches, `^32.32.32.\d{1,3}$`)
+	if testVerbose {
+		fmt.Printf("Ok. (Got active lease for MAC 0: %s)\n", lease.IpAddress)
+	}
+
+	if testVerbose {
+		fmt.Printf("# Check number of leases on Edge Gateway: ")
+	}
+	allLeases, err := edgeGateway.GetAllNsxvDhcpLeases()
+	check.Assert(err, IsNil)
+	check.Assert(allLeases, NotNil)
+	check.Assert(len(allLeases) > 0, Equals, true)
+	if testVerbose {
+		fmt.Printf("OK: (%d leases found)\n", len(allLeases))
+	}
+
+	// Check for a single NIC
+	if testVerbose {
+		fmt.Printf("# Get IP for single NIC 0: ")
+	}
+	ips, hasTimedOut, err = vm.WaitForDhcpIpByNicIndexes([]int{0}, 300, true)
+	check.Assert(err, IsNil)
+	check.Assert(hasTimedOut, Equals, false)
+	check.Assert(ips, HasLen, 1)
+	check.Assert(ips[0], Matches, `^32.32.32.\d{1,3}$`)
+	if testVerbose {
+		fmt.Printf("OK: Got IP for NICs 0: %s\n", ips[0])
+	}
+
+	// Check if IPs are reported by only using VMware tools
+	if testVerbose {
+		fmt.Printf("# Get IPs for NICs 0 and 1 (only using guest tools): ")
+	}
+	ips, hasTimedOut, err = vm.WaitForDhcpIpByNicIndexes([]int{0, 1}, 300, false)
+	check.Assert(err, IsNil)
+	check.Assert(hasTimedOut, Equals, false)
+	check.Assert(ips, HasLen, 2)
+	check.Assert(ips[0], Matches, `^32.32.32.\d{1,3}$`)
+	check.Assert(ips[1], Matches, `^32.32.32.\d{1,3}$`)
+	if testVerbose {
+		fmt.Printf("OK: IPs for NICs 0 and 1 (via guest tools): %s, %s\n", ips[0], ips[1])
+	}
+
+	// Cleanup vApp
+	err = deleteVapp(vcd, vapp.VApp.Name)
+	check.Assert(err, IsNil)
+}
+
+// getOrgVdcNetworkWithDhcp is a helper that creates a routed Org network and a DHCP pool with
+// single IP address to be assigned. Org vDC network and IP address assigned to DHCP pool are
+// returned
+func getOrgVdcNetworkWithDhcp(vcd *TestVCD, check *C, edgeGateway *EdgeGateway) (*types.OrgVDCNetwork, string) {
```

Comment:
If this function creates something, it shouldn't be called `get...`. 
Perhaps `makeOrgVdcNetworkWithDhcp` ?

Moreover, the returned IP address is hard-coded. Should we pass it as a parameter instead?

Reply:
This is actually not needed at all currently (I ignore this output using `_` assignment). This is a leftover from development phase and I think it is not worth building a more complicated function (at least at this time).

Reply:
Removed.